controllers/perconaservermysql_controller.go

```diff
@@ -1531,3 +1539,179 @@ func (r *PerconaServerMySQLReconciler) restartGroupReplication(ctx context.Conte
 
 	return nil
 }
+
+func (r *PerconaServerMySQLReconciler) createSSLByCertManager(cr *apiv1alpha1.PerconaServerMySQL) error {
+	owner, err := OwnerRef(cr, r.Scheme)
+	if err != nil {
+		return err
+	}
+	ownerReferences := []metav1.OwnerReference{owner}
+
+	issuerName := cr.Name + "-pso-issuer"
+	caIssuerName := cr.Name + "-pso-ca-issuer"
+	issuerKind := "Issuer"
+	issuerGroup := ""
+	if cr.Spec.TLS != nil && cr.Spec.TLS.IssuerConf != nil {
+		issuerKind = cr.Spec.TLS.IssuerConf.Kind
+		issuerName = cr.Spec.TLS.IssuerConf.Name
+		issuerGroup = cr.Spec.TLS.IssuerConf.Group
+	} else {
+		if err := r.createIssuer(ownerReferences, cr.Namespace, caIssuerName, ""); err != nil {
+			return err
+		}
+
+		caCert := &cm.Certificate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            cr.Name + "-ca-cert",
+				Namespace:       cr.Namespace,
+				OwnerReferences: ownerReferences,
+			},
+			Spec: cm.CertificateSpec{
+				SecretName: cr.Name + "-ca-cert",
+				CommonName: cr.Name + "-ca",
+				IsCA:       true,
+				IssuerRef: cmmeta.ObjectReference{
+					Name:  caIssuerName,
+					Kind:  issuerKind,
+					Group: issuerGroup,
+				},
+				Duration:    &metav1.Duration{Duration: time.Hour * 24 * 365},
+				RenewBefore: &metav1.Duration{Duration: 730 * time.Hour},
+			},
+		}
+
+		err = r.Create(context.TODO(), caCert)
+		if err != nil && !k8serrors.IsAlreadyExists(err) {
+			return errors.Wrap(err, "create CA certificate")
+		}
+
+		if err := r.waitForCerts(cr.Namespace, cr.Name+"-ca-cert"); err != nil {
+			return err
+		}
+
+		if err := r.createIssuer(ownerReferences, cr.Namespace, issuerName, caCert.Spec.SecretName); err != nil {
+			return err
+		}
+	}
+	hosts := []string{
+		fmt.Sprintf("*.%s-mysql", cr.Name),
+		fmt.Sprintf("*.%s-mysql.%s", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-mysql.%s.svc.cluster.local", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-orchestrator", cr.Name),
+		fmt.Sprintf("*.%s-orchestrator.%s", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-orchestrator.%s.svc.cluster.local", cr.Name, cr.Namespace),
```

Comment:
we should also include router hostnames (`<cr.Name>-router`) for group replication. I think we can include them by default, no need to check the replication type